xpdacq/beamtime.py

```diff
@@ -168,6 +168,66 @@ def Tramp(dets, exposure, Tstart, Tstop, Tstep, *, md=None):
     yield from plan
 
 
+def T_list(dets, exposure, T_list):
+    """
+    Taking images from the area detector while scaning over the temperature
+    controller in a list of user defined steps
+
+    Parameters
+    ----------
+    dets : list
+        list of objects that represent instrument devices. In xpdAcq, it is
+        defaulted to area detector.
+    exposure : float
+        total time of exposure in seconds for area detector
+    T_list : list
+        a list of temperature steps that you wish to scan over
+
+    Note
+    ----
+    area detector and temperature controller that would be triggered will
+    always be the one configured in global state. Please using following
+    commands:
+
+        >>> glbl.area_det
+        >>> glbl.temp_controller
+
+    To interrogate which devices are currently in use.
+    """
+
+    pe1c, = dets
+    # setting up area_detector and temp_controller
+    (num_frame, acq_time, computed_exposure) = _configure_pe1c(exposure)
+    motor = glbl.temp_controller
+    # manage md
+    bluesky_parent_md = {'detectors': [det.name for det in dets],
```

Comment:
All this is already provided by list_scan itself. Why copy/paste here? The only metadata you need to override is plan_name, which should match the function name (Tlist or T_list, whichever you go with).